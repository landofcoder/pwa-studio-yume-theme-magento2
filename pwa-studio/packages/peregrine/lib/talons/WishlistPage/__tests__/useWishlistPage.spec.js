import React from 'react';
import { useHistory } from 'react-router-dom';

import createTestInstance from '../../../util/createTestInstance';
import { useUserContext } from '../../../context/user';
import { useWishlistPage } from '../useWishlistPage';
import { useQuery } from '@apollo/client';

jest.mock('react-router-dom', () => ({
    useHistory: jest.fn()
}));
jest.mock('@apollo/client', () => ({
    useQuery: jest.fn().mockReturnValue({})
}));
jest.mock('../../../context/user', () => ({
    useUserContext: jest.fn().mockReturnValue([{ isSignedIn: true }])
}));
jest.mock('../../../hooks/useTypePolicies', () => ({
    useTypePolicies: jest.fn()
}));

const Component = props => {
    const talonProps = useWishlistPage(props);

    return <i talonProps={talonProps} />;
};

const props = {
    queries: {}
};

test('return correct shape', () => {
    const { root } = createTestInstance(<Component {...props} />);
    const { talonProps } = root.findByType('i').props;

    expect(talonProps).toMatchSnapshot();
});

test('unauth redirect to home', () => {
    const mockHistoryPush = jest.fn();
    useHistory.mockReturnValue({ push: mockHistoryPush });
    useUserContext.mockReturnValue([{ isSignedIn: false }]);

    createTestInstance(<Component {...props} />);

    expect(mockHistoryPush).toHaveBeenCalled();
});

test('return wishlist data', () => {
    useQuery.mockReturnValue({
        data: { customer: { wishlists: ['wishlist1', 'wishlist2'] } }
    });

    const { root } = createTestInstance(<Component {...props} />);
    const { talonProps } = root.findByType('i').props;

    expect(talonProps.wishlists).toMatchSnapshot();
});
