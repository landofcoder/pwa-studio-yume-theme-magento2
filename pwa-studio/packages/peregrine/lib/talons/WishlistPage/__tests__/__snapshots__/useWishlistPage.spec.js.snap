// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`return correct shape 1`] = `
Object {
  "errors": Map {
    "getCustomerWishlistQuery" => undefined,
  },
  "wishlists": Array [],
}
`;

exports[`return wishlist data 1`] = `
Array [
  "wishlist1",
  "wishlist2",
]
`;
